pkg/apis/config/v1alpha2/meshrootcertificate.go

```diff
@@ -111,11 +115,69 @@ type TresorCASpec struct {
 	SecretRef corev1.SecretReference `json:"secretRef"`
 }
 
-// MeshRootCertificateStatus defines the status of the MeshRootCertificate resource
+// MeshRootCertificateIntent specifies the intent of the MeshRootCertificate
+// can be (Active, Passive).
+type MeshRootCertificateIntent string
+
+// MeshRootCertificateComponentStatus specifies the status of the certificate component,
+// can be (`True`, `False`, `Unknown`).
+type MeshRootCertificateComponentStatus string
+
+// MeshRootCertificateConditionStatus specifies the status of the MeshRootCertificate condition,
+// one of (`True`, `False`, `Unknown`).
+type MeshRootCertificateConditionStatus string
+
+// MeshRootCertificateConditionType specifies the type of the condition,
+// one of (`Ready`, `Accepted`, `IssuingRollout`, `ValidatingRollout`, `IssuingRollback`, `ValidatingRollback`).
+type MeshRootCertificateConditionType string
+
+// MeshRootCertificateComponentStatuses is the set of statuses for each certificate component in the cluster.
+type MeshRootCertificateComponentStatuses struct {
+	Webhooks        MeshRootCertificateComponentStatus `json:"webhooks"`
+	XDSControlPlane MeshRootCertificateComponentStatus `json:"xdsControlPlane"`
+	Sidecar         MeshRootCertificateComponentStatus `json:"sidecar"`
+	Bootstrap       MeshRootCertificateComponentStatus `json:"bootstrap"`
+	Gateway         MeshRootCertificateComponentStatus `json:"gateway"`
+}
+
+// MeshRootCertificateCondition defines the condition of the MeshRootCertificate resource.
+type MeshRootCertificateCondition struct {
+	// Type of the condition,
+	// one of (`Ready`, `Accepted`, `IssuingRollout`, `ValidatingRollout`, `IssuingRollback`, `ValidatingRollback`).
+	Type MeshRootCertificateConditionType `json:"type"`
+
+	// Status of the condition, one of (`True`, `False`, `Unknown`).
+	Status MeshRootCertificateConditionStatus `json:"status"`
+
+	// LastTransitionTime is the timestamp corresponding to the last status
+	// change of this condition.
+	// +optional
+	LastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty"`
+
+	// Reason is a brief machine readable explanation for the condition's last
+	// transition (should be in camelCase).
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// Message is a human readable description of the details of the last
+	// transition, complementing reason.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+// MeshRootCertificateStatus defines the status of the MeshRootCertificate resource.
 type MeshRootCertificateStatus struct {
-	// State specifies the state of the certificate provider
+	// State specifies the state of the certificate provider.
 	// All states are specified in constants.go
 	State string `json:"state"`
+
+	// Set of statuses for each certificate component in the cluster (e.g. webhooks, bootstrap, etc.).
```

Comment:
nit: Maybe remove the period from the end of the comment to be consistent with the rest of the file.
```suggestion
	// Set of statuses for each certificate component in the cluster (e.g. webhooks, bootstrap, etc.)
```